src/shims/unix/linux/eventfd.rs

```diff
@@ -182,9 +182,6 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
     fn eventfd(&mut self, val: &OpTy<'tcx>, flags: &OpTy<'tcx>) -> InterpResult<'tcx, Scalar> {
         let this = self.eval_context_mut();
 
-        // eventfd is Linux specific.
-        this.assert_target_os("linux", "eventfd");
-
```

Comment:
If the intend is to only eventfd on linux and android only, instead of removing the assert, maybe we could use

```rust
if !matches!(&*this.tcx.sess.target.os, "linux" | "android") {
     panic!("`eventfd` should not be called on {}", this.tcx.sess.target.os);
}
```

Reply:
The thing here is that the `eventfd` implementation resides in the `linux` module, so I see no reason in that assertion. If it would be for any `unix` with some exceptions, then yes, there's a good reason to keep the assertion.

Reply:
@RalfJung, what's your opinion?

Reply:
Seems like maybe it's a good time to introduce a `unix::linux_like` module, and then put some things there that both `unix::android` and `unix::linux` can import and use as helpers.

I agree that in that module, if things work on all linux-like systems, they don't need a further assertion.